Assets/Scripts/ECS/Animal/EnergySystem.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Unity.Entities;
+using Ecosystem.ECS.Death;
+using Ecosystem.ECS.Movement;
+
+namespace Ecosystem.ECS.Animal {
+    /// <summary>
+    /// System for updating the energy of an animal.
+    /// </summary>
+    public class EnergySystem : SystemBase {
+        private float recoveryTime;
+        private float recoveryLimit;
+        private EndSimulationEntityCommandBufferSystem m_EndSimulationEcbSystem;
+
+        protected override void OnCreate() {
+            recoveryTime = 0f;
+            recoveryLimit = 0.5f;
+            m_EndSimulationEcbSystem = World
+                .GetOrCreateSystem<EndSimulationEntityCommandBufferSystem>();
+        }
+
+        protected override void OnUpdate() {
+            var commandBuffer = m_EndSimulationEcbSystem.CreateCommandBuffer().ToConcurrent();
+
+            float deltaTime = Time.DeltaTime;
+
+            Entities.ForEach((Entity entity,EntityManager entityManager,int entityInQueryIndex,
```

Comment:
```suggestion
            Entities.ForEach((Entity entity,int entityInQueryIndex,
```
EntityManager kan inte vara en parameter här. Man använder propertyn 'EntityManager' direkt istället.